### PR TITLE
start to replace aws-sdk v2

### PR DESCRIPTION
**Pull request by aochsner**
also w/ more robust s3 object url handling

#16470 

## Hey, I just made a Pull Request!

Replacing v2 `aws-sdk` with v3 in `plugins-catalog-backend-module-aws`

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by andrewthauer** [comment]
@aochsner - Have you seen the newish [integration-aws-node package](https://github.com/backstage/backstage/blob/master/packages/integration-aws-node/README.md)? Might make sense to leverage this for the credentials portion.

**Comment by pjungermann** [comment]
I agree with @andrewthauer. The integration-aws-node package should be used here.

**Comment by aochsner** [comment]
Makes sense.  Used on my other PR #17619...

**Comment by aochsner** [comment]
Ok should be good to go.  AWS sdk versions should be the same so there isn't type confusion.  AWS sdk doesn't follow semantic-versioning.  Not sure if there's a way to simplify this.

File: plugins/catalog-backend-module-aws/api-report.md [hunk 1]
```diff
@@ -42,9 +48,9 @@ export class AwsOrganizationCloudAccountProcessor implements CatalogProcessor {
   static fromConfig(
     config: Config,
     options: {
-      logger: Logger;
+      logger?: Logger;
```
Comment (freben) [inline plugins/catalog-backend-module-aws/api-report.md:51]: Was this change made for some particular reason?

File: plugins/catalog-backend-module-aws/api-report.md [hunk 2]
```diff
@@ -19,11 +20,16 @@ import { UrlReader } from '@backstage/backend-common';
 // @public
 export type AWSCredentialFactory = (
   awsAccountId: string,
-) => Promise<Credentials>;
+) => Promise<AwsCredentialIdentity>;
```
Comment (freben) [inline plugins/catalog-backend-module-aws/api-report.md:23]: This is technically breaking, but they are very alike aren't they? I think we can let that one slide potentially.

File: plugins/catalog-backend-module-aws/api-report.md [hunk 3]
```diff
@@ -42,9 +48,9 @@ export class AwsOrganizationCloudAccountProcessor implements CatalogProcessor {
   static fromConfig(
     config: Config,
     options: {
-      logger: Logger;
+      logger?: Logger;
     },
-  ): AwsOrganizationCloudAccountProcessor;
+  ): Promise<AwsOrganizationCloudAccountProcessor>;
```
Comment (freben) [inline plugins/catalog-backend-module-aws/api-report.md:53]: This one is breaking though. Might want to upgrade this to a `minor` changeset then.

**Review by freben — COMMENTED** [review]
Super nice! Just some small things.

File: plugins/catalog-backend-module-aws/api-report.md (hunk 2 above)
Comment (aochsner) [reply plugins/catalog-backend-module-aws/api-report.md:23]: Correct.  `Credentials` appears to have behavior (although Backstage doesn't depend on it) whereas `AwsCredentialIdentity` does not.

File: plugins/catalog-backend-module-aws/api-report.md (hunk 3 above)
Comment (aochsner) [reply plugins/catalog-backend-module-aws/api-report.md:53]: Yep good catch

File: plugins/catalog-backend-module-aws/api-report.md (hunk 1 above)
Comment (aochsner) [reply plugins/catalog-backend-module-aws/api-report.md:51]: I don't think so after trying to dig through it a little bit

**Comment by aochsner** [comment]
github actions have been having a rough day.... will bump this tomorrow (any easier way to do that other than close/reopen the PR?)

File: .changeset/itchy-impalas-fold.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+'@backstage/plugin-events-backend-module-aws-sqs': patch
+'@backstage/plugin-catalog-backend-module-aws': minor
```
Comment (freben) [inline .changeset/itchy-impalas-fold.md:3]: Alright it's all looking good but just one small ask. When making a breaking change, if you look at it from the user's standpoint, it's helpful to make a separate changeset for this one and declaring the breakage explicitly. Something like 

```
**BREAKING**: SomeClass.someMethod now returns a promise instead of the instance directly.
```

Happy to 🚢 after this!

**Review by freben — APPROVED** [review]
Woo!
